src/Agent.Listener/Configuration/AutoLogonConfigurationManager.cs

```diff
@@ -0,0 +1,283 @@
+#if OS_WINDOWS
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Principal;
+using System.Threading;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+{
+    [ServiceLocator(Default = typeof(AutoLogonConfigurationManager))]
+    public interface IAutoLogonConfigurationManager : IAgentService
+    {
+        void Configure(CommandSettings command);
+        void UnConfigure();
+        bool RestartNeeded();
+        bool IsAutoLogonConfigured();
```

Comment:
Not written the code but following is there in my mind.
1. use FetchAutoLogonUserDetails to know user and domain
2. make GetRegistry call for the user 
rest remains same.

---
In reply to: [118072386](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r118072386) [](ancestors = 118072386)